### PR TITLE
Add eslint and prettier

### PR DESCRIPTION
**Pull request by makylfang**

### CODE REVIEW

File: .prettierrc
Comment (karthik-durai) [inline .prettierrc]: shall we set `semi` to `false`? it makes code bit cleaner.

Comment (makylfang) [reply .prettierrc]: ok sure

File: .eslintrc.json
Comment (ajithranka) [inline .eslintrc.json]: Instead of relying on editor plugins for linting, we should install eslint and related packages as dev dependencies. 

```
yarn add --dev eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin
```

We'll also need to include `eslint-config-prettier` for handling conflicts between prettier and eslint,

```
yarn add --dev eslint-config-prettier
```

and include it in extends,

```
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier"
  ],
```
 
We can add a lint script to the package.json:

```
"lint": "eslint --ext .ts src"
```

Running this script lists 37 errors (mostly `any` type annotations) that will need to be fixed.

File: .prettierrc
Comment (ajithranka) [inline .prettierrc]: Same as with eslint, we should add prettier as a dev dependency,

```
yarn add --dev prettier
```

and include a script in package.json for formatting:

```
"format": "prettier --ignore-path .gitignore --write ."
```

Running this script should reformat all files in the project including these config files.

File: .prettierrc
Comment (ajithranka) [inline .prettierrc]: Most of these options are default in Prettier. We can create a much small config file including only the differences:

```
{
  "semi": false,
  "singleQuote": true
}
```

**Review by ajithranka — CHANGES_REQUESTED** [review]

File: .eslintrc.json
Comment (karthik-durai) [reply .eslintrc.json]: also we can use `husky` and `lint-staged`. @ajithranka

**Comment by makylfang** [comment]
This commit [778bd14](https://github.com/arcana-network/wallet/pull/8/commits/778bd1462c6c330248ecd8b7d9ebade0d21bac51) has the changes suggested, I forgot to update the commit message. 

- Added eslint, prettier, husky and lint-staged to dev dependency
- Removed wallet disabled type
- Added type for createDomElement function
- Added jest and @types/jest

File: .eslintcache
Comment (ajithranka) [inline .eslintcache:1]: This auto-generated file probably needs to be added to `.gitignore`.

**Review by ajithranka — APPROVED** [review]

File: .eslintcache
Comment (makylfang) [reply .eslintcache:1]: Yeah the last commit is doing that. [3227b5f](https://github.com/arcana-network/wallet/pull/8/commits/3227b5ff2aea8b4e139acdec970f2883625483bf)

Will delete file also it in the next commit so it doesn't shows up in the repo.

**Comment by makylfang** [comment]
@karthik-durai Can you also check this once? I'll merge it after that.

**Pull request merged**
